addons/stock/models/stock_picking.py

```diff
@@ -1514,15 +1514,17 @@ def _package_move_lines(self, batch_pack=False):
                 float_compare(ml.quantity, 0.0, precision_rounding=ml.product_uom_id.rounding) > 0 and
                 not ml.result_package_id
         )
+        if move_lines:
+            quantity_move_line_ids = quantity_move_line_ids & move_lines
         move_line_ids = quantity_move_line_ids.filtered(lambda ml: ml.picked)
         if not move_line_ids:
             move_line_ids = quantity_move_line_ids
         return move_line_ids
 
-    def action_put_in_pack(self):
+    def action_put_in_pack(self, move_lines=False):
```

Comment:
You can't modify a method signature in stable (if somebody developed extension using this method, you could eventually break it.)
I think for now, you have no other solution than creating another version of `action_put_in_pack` for when you want to do it on a subset of move lines and wait for the master's forward-port to do it properly

Reply:
Even if it's a keyword argument?

Reply:
Yep.
Also, I said there is no other solution than creating another version of `action_put_in_pack` but it's not true, you can also give the move lines' ids to the context and use it in `_package_move_lines` to get only the move lines you want, it's maybe easier and cleaner for now (sorry for not thinking of that earlier, I saw you already pushed some changes...)

Reply:
Ah okay, that's much cleaner indeed. I just pushed again 👍

Reply:
BUT, it's only on stable version. Once it arrives in master, it's a good practice to avoid context since it's not explicit the way they are defined and used. So you can put back the kwargs in master. It's a bit annoying and not productive to do the development twice, but it's the price to avoid breaking community custom on stable version